persistence/javadsl/src/main/scala/com/lightbend/lagom/javadsl/persistence/PersistentEntity.scala

```diff
@@ -218,30 +216,75 @@ abstract class PersistentEntity[Command, Event, State] {
       _state = state
     }
 
+    def setEventHandler[A <: Event](eventClass: Class[A], handler: JFunction[A, State]): Unit =
+      setEventHandlerChangingBehavior[A](eventClass, new JFunction[A, Behavior] {
+        override def apply(evt: A): Behavior = behavior.withState(handler.apply(evt))
+      })
+
+    def setEventHandlerChangingBehavior[A <: Event](eventClass: Class[A], handler: JFunction[A, Behavior]): Unit =
+      eventHandlers = eventHandlers.updated(eventClass, handler)
+
+    def removeEventHandler(eventClass: Class[_ <: Event]): Unit =
+      eventHandlers -= eventClass
+
+    def setCommandHandler[R, A <: Command with ReplyType[R]](
+      commandClass: Class[A],
+      handler:      JBiFunction[A, CommandContext[R], Persist[_ <: Event]]
+    ): Unit = {
+      commandHandlers = commandHandlers.updated(
+        commandClass,
+        handler.asInstanceOf[JBiFunction[A, CommandContext[Any], Persist[_ <: Event]]]
+      )
+    }
+
+    def removeCommandHandler(commandClass: Class[_ <: Command]): Unit =
+      commandHandlers -= commandClass
+
+    def setReadOnlyCommandHandler[R, A <: Command with ReplyType[R]](
+      commandClass: Class[A],
+      handler:      JBiConsumer[A, ReadOnlyCommandContext[R]]
+    ): Unit = {
+      setCommandHandler[R, A](commandClass, new JBiFunction[A, CommandContext[R], Persist[_ <: Event]] {
+        override def apply(cmd: A, ctx: CommandContext[R]): Persist[Event] = {
+          handler.accept(cmd, ctx)
+          ctx.done()
+        }
+      });
+    }
+
+    def build(): Behavior =
+      Behavior(_state, eventHandlers, commandHandlers)
+
+  }
+
+  /**
+   * Mutable builder that is used for defining the event and command handlers.
+   * Use [#build] to create the immutable [[PersistentEntity.Behavior]].
+   */
+  // In order to provide javadoc preventing isntantiantiation or extension this sealed trait is added
+  // to hold docs and BehaviorBuilderImpl is made private to hold implementation
+  sealed trait BehaviorBuilder {
+    def getState(): State
+    def setState(state: State): Unit
     /**
      * Register an event handler for a given event class. The `handler` function
      * is supposed to return the new state after applying the event to the current state.
      * Current state can be accessed with the `state` method of the `PersistentEntity`.
      */
```

Comment:
It would be nice to clarify that each of the `set` methods overrides any previously-set handler for that class. Maybe obvious to some, but not all. Related to #1166.